src/service/angular-sails.js

```diff
@@ -1,74 +1,168 @@
+        if(provider.debug) $log.info('$sails ' + config.method.toUpperCase() + ' ' + config.url, config.data || '');
+
+        socket['legacy_' + config.method](config.url, config.data, function(result, jwr) {
+          // resolve promise if JSON web response is an object that has a statusCode 2xx
+          jwr.data = result; // backward compat, jwr.body also holds your data
+          jwr.socket = socket;
+          jwr.url = config.url;
+          jwr.method = config.method;
+          jwr.config = config.config;
+          if(!jwr || !angular.isObject(jwr) || !jwr.statusCode || jwr.statusCode < 200 || jwr.statusCode >= 300) {
+            if(provider.debug) $log.warn('$sails response ' + jwr.statusCode + ' ' + config.url, jwr);
+            defer.reject(jwr);
+          } else {
+            if(provider.debug) $log.info('$sails response ' + config.url, jwr);
+            defer.resolve(jwr);
+          }
+        });
+        return defer.promise;
+      }
+
+      // Wrap a socket.io method within the promis chain
+      function promisify(methodName) {
+        socket['legacy_' + methodName] = socket[methodName];
+
+        socket[methodName] = function(url, data, config) {
+
+          var chain = [serverRequest, undefined];
+          var promise = $q.when({
+            url: provider.base + url,
+            data: data,
+            socket: socket,
+            config: config || {},
+            method: methodName
+          });
+
+          // apply interceptors
+          angular.forEach(reversedInterceptors, function(interceptor) {
+            if(interceptor.request || interceptor.requestError) {
+              chain.unshift(interceptor.request, interceptor.requestError);
+            }
+            if(interceptor.response || interceptor.responseError) {
+              chain.push(interceptor.response, interceptor.responseError);
+            }
+          });
+
+          while(chain.length) {
+            var thenFn = chain.shift();
+            var rejectFn = chain.shift();
+
+            promise = promise.then(thenFn, rejectFn);
+          }
+
+          return promise;
+        };
+      }
+
+
+      // Wrap events to ensure a $digest cycle
+      function wrapEvent(eventName) {
+        socket['legacy_' + eventName] = socket[eventName];
+        socket[eventName] = function(event, cb) {
+          if(cb !== null && angular.isFunction(cb)) {
+            socket['legacy_' + eventName](event, function(result) {
+              $rootScope.$evalAsync(cb.bind(socket, result));
+            });
+          }
+        };
+      }
+
+
+      angular.forEach(provider.httpVerbs, promisify);
+      angular.forEach(provider.eventNames, wrapEvent);
+
+
+      /**
+       * Update a model on sails pushes
+       * @param {string} name       Sails model name
+       * @param {objcet} modelObj   angular model object
+       * @todo remove the lodash dependency
+       */
+      socket.$modelUpdater = function(name, modelObj) {
```

Comment:
I like this, but see if we can do without `_`.  If not, include it as a dependency and note it in the README.